Orange/widgets/data/owfeatureconstructor.py

```diff
@@ -115,8 +121,11 @@ def __init__(self, *args, **kwargs):
                                    QSizePolicy.Fixed)
         )
         self.expressionedit = QLineEdit(
-            placeholderText="Expression..."
-        )
+            placeholderText="Expression...",
+            toolTip="Result must be a number for numeric variables, "
+                    "and strings for text variables.\n"
```

Comment:
I agreed, implemented it, and no longer agree. It looks visually as the same line edit and the user who will get a tooltip once (and the default is that for numeric attributes, which tells the user to enter a numeric expression) will not expect to get a different tooltip the next time.

Please check and tell me that I can/should remove 09bfee3b1. :)

Reply:
I agree it is not perfect, but I think removing ac97c90 is worse... The best thing about the tooltip for datetime is that you were able to put in an example, so people actually know what they have to produce in the expression. If those 3 lines are added to the original 4 of the tooltip it becomes a mess.
And anyway, I think the problem would be minimal if when no variable has yet been selected it showed something generic or even no tooltip.
But until the next round of improvements, I would say let's just keep this commit. People not expecting a different tooltip is probably not that big a problem, because you usually need to check it when you have an error in the expression and then you see the relevant information.